api/src/main/java/net/md_5/bungee/api/event/ServerConnectFailEvent.java

```diff
@@ -0,0 +1,37 @@
+package net.md_5.bungee.api.event;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import net.md_5.bungee.api.ServerConnectRequest;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.api.connection.Server;
+import net.md_5.bungee.api.plugin.Event;
+
+/**
+ * This event is called when a connection to a server fails.
+ * This can occur when the server is offline, the player cannot login due to version mismatch, or the server is full.
```

Comment:
version mismatch and server is full is jsut server kicked player

Reply:
Sorry, this description was incorrect.  
This event only fires when the connection itself fails, such as when the server is offline.  
It does not fire when version mismatch or server is full, so I have fixed documentation.